### PR TITLE
feat: human readable urls & quicksearch

### LINKED ISSUE #342: [Feature]: More readable URLs
### What feature do you want to add?

More readable urls, for example: `/media/JW-1234--shrek-3`.

### Why do you want to have this feature?

Allows people to see the title of the movie in the url, can be a good alternative instead of meta tags, which we aren't able to do due to being a client-only web application.

### Anything other details to share?

the title in the url would be purely cosmetic. it does not change any functionality or have any validation when loading it in.

### LINKED ISSUE #349: [Feature]: Quick search
### What feature do you want to add?

A new endpoint: `/s/:query`.
it should search and immediately watch the first result.

### Why do you want to have this feature?

Makes it quicker to search and watch.

### Anything other details to share?

_No response_

### PR DESCRIPTION
**Pull request by castdrian**
This pr closes #342 and also closes #349 by implementing the following:

- urls get the slugified media title appended, which is purely cosmetic
`/media/tmdb-show-103516` -> `/media/tmdb-show-103516-star-trek-strange-new-worlds`

- new endpoint /s/:query automatically searches and plays the first result for the provided query
`/s/smallville` -> `/media/tmdb-show-4604-smallville`

### CODE REVIEW

**Review by binaryoverload — APPROVED** [review]
This looks good to me, it just needs testing by us.

File: src/backend/metadata/getmeta.ts [hunk 1]
```diff
@@ -224,10 +204,17 @@ export async function convertLegacyUrl(
...
+        strict: true,
+      })}`;
   }
 
   if (tmdbId) {
-    return `/media/tmdb-${type}-${tmdbId}`;
+    return `/media/tmdb-${type}-${tmdbId}-${slugify(meta.meta.title, {
+      lower: true,
+      strict: true,
+    })}`;
```
Comment (mrjvs) [inline src/backend/metadata/getmeta.ts:218]: I think you should have a function for this instead. Something like `TMDBIdToUrlId(type, tmdbId, title)`.
You should then also use it in `TMDBMediaToId()`. So there is only one place that makes urls, makes it easier to change later

**Review by mrjvs — CHANGES_REQUESTED** [review]
Just one tiny thing, for the rest it looks good. nice job.

File: src/backend/metadata/getmeta.ts (hunk 1 above)
Comment (castdrian) [reply src/backend/metadata/getmeta.ts:218]: good catch, will do :)

**Review by mrjvs — APPROVED** [review]

**Pull request merged**